src/vs/workbench/contrib/terminal/browser/links/links.ts

```diff
@@ -0,0 +1,63 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IBufferRange, Terminal } from 'xterm';
+
+export interface ITerminalLinkDetector {
+	/**
+	 * The xterm.js instance this detector belongs to.
+	 */
+	readonly xterm: Terminal;
+
+	/**
+	 * Detects links within the _wrapped_ line range provided and returns them as an array.
+	 *
+	 * @param startLine The start of the wrapped line. This _will not_ be validated that it is
+	 * indeed the start of a wrapped line.
```

Comment:
`This _will not_ be validated that it is indeed the start of a wrapped line.`

❓

Reply:
Meaning make sure it's actually a wrapped line because passing in startLine/endLine because this function doesn't verify it